File: client/src/app/entities/purchase.ts
import {User} from "./user";
import {Supplier} from "./supplier";
import {DataPage} from "../shared/data-page";
import {Purchaseitem} from "./purchaseitem";

export class Purchase{
  id: number;
  code: string;
  tocreation: string;
  description: string;
  creator: User;
  supplier: Supplier;
  date: string;
  total: number;

  purchaseitemList: Purchaseitem[];

  constructor(id: number) {
    this.id = id;
  }
}
export class PurchaseDataPage extends DataPage{
  content: Purchase[];
}

File: server/src/main/java/bit/project/server/entity/Purchase.java
package bit.project.server.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
public class Purchase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDateTime tocreation;

    @Lob
    private String description;

    @ManyToOne
    @JsonIgnoreProperties({"creator", "status", "tocreation", "roleList"})
    private User creator;

    @ManyToOne
    private Supplier supplier;

    private LocalDate date;

    private BigDecimal total;

    @OneToMany(mappedBy = "purchase",cascade = CascadeType.ALL ,orphanRemoval = true)
    private List<Purchaseitem> purchaseitemList;

    public Purchase(Integer id) {
        this.id = id;
    }

    public Purchase(Integer id, String code) {
        this.id = id;
        this.code = code;
    }

}
